client/lib/i18n-utils/switch-locale.js

```diff
@@ -54,16 +54,29 @@ export default function switchLocale( localeSlug ) {
 			const directionFlag = language.rtl ? '-rtl' : '';
 			const debugFlag = process.env.NODE_ENV === 'development' ? '-debug' : '';
 			const cssUrl = window.app.staticUrls[ `style${ debugFlag }${ directionFlag }.css` ];
+
 			switchCSS( 'main-css', cssUrl );
 		}
 	} );
 }
 
+const bundles = {};
+
 export function switchCSS( elementId, cssUrl, callback = noop ) {
+	if ( bundles.hasOwnProperty( elementId ) && bundles[ elementId ] === cssUrl ) {
+		callback();
+
+		return;
+	}
```

Comment:
This is to handle the case where this function is called in a very short time interval, which sometimes leads to injecting the same `<link />` element twice. This particular operation indeed isn't instantaneous, and as such the test below that checks for the presence of this element will fail while one is already in the process of being injected in the DOM.

Reply:
Oh I see, because the `id` attribute is set asynchronously... I think we should change this to set it synchronously by default except when it already exists.